pycript/Reqcheck.py

```diff
@@ -204,40 +191,33 @@ def encrypt_and_update_parameters(extender, currentreq, encryptionpath, selected
         
         else:
-            if param.getType() == IParameter.PARAM_URL:
-                decrypteedparam =  Parameterencrypt(selectedlang, encryptionpath, param.getValue())
-                currentreq = extender.helpers.updateParameter(currentreq, extender.helpers.buildParameter(param.getName(), decrypteedparam, param.getType()))
-
-            elif param.getType() == IParameter.PARAM_BODY:
-                decrypteedparam =  Parameterencrypt(selectedlang, encryptionpath, param.getValue())
-                currentreq = extender.helpers.updateParameter(currentreq, extender.helpers.buildParameter(param.getName(), decrypteedparam, param.getType()))
-
-    parameters = extender.helpers.analyzeRequest(currentreq).getParameters()
-    header = extender.helpers.analyzeRequest(currentreq).getHeaders()
+            print("Something went wrong, Please check the request and logger")
 
     for param in parameters:
         if selected_method == "BOTH" and param.getType() == IParameter.PARAM_JSON:
             json_object = loads(body)
             json_object = update_json_value(json_object, selectedlang, encryptionpath, Parameterencrypt,selected_request_inc_ex_ctype,listofparam)
             output = extender.helpers.stringToBytes(dumps(json_object,separators=(',', ':')))
             currentreq = extender.helpers.buildHttpMessage(header, output)
-            break
+        elif selected_method == "BOTH" and param.getType() != IParameter.PARAM_JSON:
+            decrypteedparam =  update_raw_value(param, selectedlang, encryptionpath, Parameterencrypt,selected_request_inc_ex_ctype,listofparam)
+            currentreq = extender.helpers.updateParameter(currentreq, extender.helpers.buildParameter(param.getName(), decrypteedparam, param.getType()))
+        else:
+            print("Something went wrong, Please check the request and logger")
 
     return currentreq
 
 def encrypt_and_update_parameter_keys_and_values(extender, currentreq, encryptionpath, selected_method, selectedlang, body, parameters, header,selected_request_inc_ex_ctype,listofparam):
     for param in parameters:
         if selected_method == "GET" and param.getType() == IParameter.PARAM_URL:
-            encrypted_param_name = Parameterencrypt(selectedlang, encryptionpath, param.getName())
-            encrypted_param_value = Parameterencrypt(selectedlang, encryptionpath, param.getValue())
+            encrypted_param_name, encrypted_param_value = update_raw_key_value(param, selectedlang, encryptionpath, Parameterencrypt, selected_request_inc_ex_ctype, listofparam)
             currentreq = extender.helpers.removeParameter(currentreq, param)
             new_param = extender.helpers.buildParameter(encrypted_param_name, encrypted_param_value, param.getType())
             currentreq = extender.helpers.addParameter(currentreq, new_param)
 
         elif selected_method == "BODY" and param.getType() != IParameter.PARAM_URL:
             if param.getType() == IParameter.PARAM_BODY:
```

Comment:
same, the loop is not required for JSON 

the `selected_method == "BOTH" and param.getType() != IParameter.PARAM_JSON` 
the BURP API also has XML and other types, the Not ` IParameter.PARAM_JSON` will be triggered for parameter types that are not supported like XML even cookies as well, multi-part which can be image or file upload data